packages/plugin-ethstorage/src/actions/submitData.ts

```diff
@@ -0,0 +1,207 @@
+import { ethstorageAvailConfig } from "../environment";
+import { BlobUploader } from "../utils/uploader";
+import { encodeOpBlobs } from "../utils/blobs.ts";
+
+export const EthStorageAbi: readonly string[] = [
+    'function putBlobs(bytes32[] memory _keys, uint256[] memory _blobIdxs, uint256[] memory _lengths)',
+    'function putBlob(bytes32 _key, uint256 _blobIdx, uint256 _length) public payable',
+    'function get(bytes32 _key, uint8 _decodeType, uint256 _off, uint256 _len) public view returns (bytes memory)',
+    'function size(bytes32 _key) public view returns (uint256)',
+    'function upfrontPayment() public view returns (uint256)'
+];
+
+export function stringToHex(s: string): string {
+    return ethers.hexlify(ethers.toUtf8Bytes(s));
+}
+
+export async function sendData(
+    RPC: string, privateKey: string,
+    address: string, key: string, data: Buffer
+): Promise<string> {
+    const blobUploader = await BlobUploader.create(RPC, privateKey);
+    const provider = new ethers.JsonRpcProvider(RPC);
+    const wallet = new ethers.Wallet(privateKey, provider);
+    const contract = new ethers.Contract(address, EthStorageAbi, wallet);
+    const hexKey = ethers.keccak256(stringToHex(key));
+    const storageCost = await contract.upfrontPayment();
+    const tx = await contract.putBlob.populateTransaction(hexKey, 0, data.length, {
+        value: storageCost,
+    });
+
+    const blobs = encodeOpBlobs(data);
+    const txRes = await blobUploader.sendTx(tx, blobs);
+    elizaLogger.log(`tx hash ${txRes.hash}`);
+    const receipt = await txRes.wait();
+    return receipt?.status === 1 ? txRes.hash : undefined;
+}
+
+const submitDataTemplate = `Respond with a JSON markdown block containing only the extracted values. Use null for any values that cannot be determined.
+
+
+Example response:
+\`\`\`json
+{
+    "key": "data_key"
+    "data": "Hello World, this is the data I submitted",
+}
+\`\`\`
+
+{{recentMessages}}
+
+Given the recent messages, extract the following information about the requested EthStorage data submission:
+- Key of the data to be submitted
+- Data to be submitted
+
+Respond with a JSON markdown block containing only the extracted values.`;
+
+export default {
+    name: "UPLOAD_DATA",
+    similes: [
+        "UPLOAD_DATA_TO_ETHSTORAGE",
+        "SUBMIT_DATA",
+        "SUBMIT_DATA_TO_ETHSTORAGE",
+        "SEND_DATA",
+        "SEND_DATA_TO_ETHSTORAGE",
+        "POST_DATA",
+        "POST_DATA_TO_ETHSTORAGE",
+        "POST_DATA_ON_ETHSTORAGE_NETWORK",
+        "POST_DATA_TO_ETHSTORAGE_NETWORK",
+        "SEND_DATA_ON_ETHSTORAGE_NETWORK",
+        "SEND_DATA_TO_ETHSTORAGE_NETWORK",
+        "SUBMIT_DATA_ON_ETHSTORAGE_NETWORK",
+        "SUBMIT_DATA_TO_ETHSTORAGE_NETWORK",
+        "UPLOAD_DATA_ON_ETHSTORAGE_NETWORK",
+        "UPLOAD_DATA_TO_ETHSTORAGE_NETWORK",
+    ],
+    validate: async (runtime: IAgentRuntime, _message: Memory) => {
+        await ethstorageAvailConfig(runtime);
```

Comment:
Do we want to add some validation logic here? otherwise, we will not need call ethstorageAvailConfig function?

Reply:
This function checks whether the parameter has a value, and throws an error if not.